packages/client/src/objectSet/ObjectSetListenerWebsocket.test.ts

```diff
@@ -126,8 +135,15 @@ describe("ObjectSetListenerWebsocket", async () => {
         objectSetExpiryMs: OBJECT_SET_EXPIRY_MS,
       });
 
+      listenerPromise = new Promise((resolve) => {
+        listenerPromiseResolveFn = resolve;
+      });
```

Comment:
I'm sure we use this pattern elsewhere but could we switch to https://www.npmjs.com/package/p-defer

Reply:
Isn't this implementation better here since we can await directly on the listener which should resolve pretty quickly?